sklearn/svm/_libsvm.pyx

```diff
@@ -167,22 +173,44 @@ def fit(
     if len(sample_weight) == 0:
         sample_weight = np.ones(X.shape[0], dtype=np.float64)
     else:
-        assert sample_weight.shape[0] == X.shape[0], \
-               "sample_weight and X have incompatible shapes: " + \
-               "sample_weight has %s samples while X has %s" % \
-               (sample_weight.shape[0], X.shape[0])
+        assert (
+            sample_weight.shape[0] == X.shape[0],
+            "sample_weight and X have incompatible shapes: " +
+            "sample_weight has %s samples while X has %s" % (sample_weight.shape[0], X.shape[0])
+        )
 
     kernel_index = LIBSVM_KERNEL_TYPES.index(kernel)
     set_problem(
-        &problem, X.data, Y.data, sample_weight.data, X.shape, kernel_index)
+        &problem,
+        <char*> &X[0, 0],
+        <char*> &Y[0],
+        <char*> &sample_weight[0],
+        <cnp.npy_intp*>X.shape,
+        kernel_index,
+    )
     if problem.x == NULL:
         raise MemoryError("Seems we've run out of memory")
-    cdef cnp.ndarray[cnp.int32_t, ndim=1, mode='c'] \
-        class_weight_label = np.arange(class_weight.shape[0], dtype=np.int32)
+    cdef cnp.int32_t[::1] class_weight_label = np.arange(class_weight.shape[0], dtype=np.int32)
     set_parameter(
-        &param, svm_type, kernel_index, degree, gamma, coef0, nu, cache_size,
-        C, tol, epsilon, shrinking, probability, <int> class_weight.shape[0],
-        class_weight_label.data, class_weight.data, max_iter, random_seed)
+        &param,
+        svm_type,
+        kernel_index,
+        degree,
+        gamma,
+        coef0,
+        nu,
+        cache_size,
+        C,
+        tol,
+        epsilon,
+        shrinking,
+        probability,
+        <int> class_weight.shape[0],
+        <char*> &class_weight_label[0] if class_weight_label.size > 0 else NULL,
+        <char*> &class_weight[0] if class_weight.size > 0 else NULL,
```

Comment:
@jjerphan The errors that are occurring on the CI are generally related to index errors on memory views which are probably empty. Over here I have handled two of such errors but now the error is occurring on line 238 `copy_sv_coef (<char*> &sv_coef[0, 0], model)`. Is there any better way to handle this or do we have to check if the memory view is empty and assign a NULL pointer as I have done here?

Reply:
I cannot think of a better way of handling it than passing `NULL`.

Reply:
Understood. Thank you for the clarification.